report-viewer/src/model/factories/ComparisonFactory.ts

```diff
@@ -98,33 +139,44 @@ export class ComparisonFactory {
     return acc
   }
 
-  private static generateColorsForMatches(num: number): Array<string> {
-    const colors: Array<string> = []
-    const numberOfColorsInFirstInterval = Math.round(((80 - 20) / (80 - 20 + (340 - 160))) * num) // number of colors from the first interval
-    const numberOfColorsInSecondInterval = num - numberOfColorsInFirstInterval // number of colors from the second interval
-    ComparisonFactory.generateColorsForInterval(20, 80, numberOfColorsInFirstInterval, colors)
-    ComparisonFactory.generateColorsForInterval(160, 340, numberOfColorsInSecondInterval, colors)
-    return colors
+  private static getSubmissionFileListFromLocal(submissionId: string): string[] {
+    const request = new XMLHttpRequest()
+    request.open('GET', `/files/submissionFileIndex.json`, false)
+    request.send()
+    if (request.status == 200) {
+      return JSON.parse(request.response).submission_file_indexes[submissionId].map(
+        (file: string) => slash(file)
+      )
+    } else {
+      return []
```

Comment:
Is this enough errors handling?

Reply:
This error handling results in not showing any files in the end.
But this error should only occur when the report was generated earlier than the current version or the file was deleted/moved